app/src/main/java/de/westnordost/streetcomplete/quests/QuestUtil.kt

```diff
@@ -29,16 +30,35 @@ fun Resources.getHtmlQuestTitle(questType: QuestType<*>, element: Element?, feat
 private fun getTemplateArguments(
     questType: QuestType<*>,
     element: Element?,
-    locale: Locale,
+    localeList: LocaleListCompat,
     featureDictionaryFuture: FutureTask<FeatureDictionary>?
 ): Array<String> {
     val tags = element?.tags ?: emptyMap()
-    val typeName = lazy { featureDictionaryFuture?.get()?.let { dict ->
-        dict.byTags(tags).forLocale(locale).find()?.firstOrNull()?.name
-    }}
+    val typeName = lazy { findTypeName(tags, featureDictionaryFuture, localeList) }
     return ((questType as? OsmElementQuestType<*>)?.getTitleArgs(tags, typeName)) ?: emptyArray()
 }
 
+private fun findTypeName(
+    tags: Map<String, String>,
+    featureDictionaryFuture: FutureTask<FeatureDictionary>?,
+    localeList: LocaleListCompat
+): String? {
+    featureDictionaryFuture?.get()?.let { dict ->
+        localeList.forEach { locale ->
+            val matches = dict.byTags(tags).forLocale(locale).find()
+            if (matches.isNotEmpty()) {
+                return matches.first().name
+            }
+        }
+    }
+    return null
+}
+
+private inline fun LocaleListCompat.forEach(action: (Locale) -> Unit) {
+    for (i in 0 until size()) {
+        action(this[i])
+    }
+}
```

Comment:
Wow sheesh, Google devs did not even implement the `Iterable` interface?

Reply:
Anyway, looks good!

Maybe since this is actually a new implementation, you could test this once? UnitTest would of course be supreme, but not required, there wasn't any before anyway.

Reply:
The method is called for the "What is the name of this place?" quest.

Reply:
> 
> 
> Wow sheesh, Google devs did not even implement the `Iterable` interface?

Yeah, it seems to be the case with the actual [`LocaleList`](https://developer.android.com/reference/android/os/LocaleList.html) as well.

Reply:
> 
> 
> Anyway, looks good!
> 
> Maybe since this is actually a new implementation, you could test this once? UnitTest would of course be supreme, but not required, there wasn't any before anyway.

Sure.

Reply:
There isn't any visible difference, as far as I can tell.

![Screenshot_20200819-184829226](https://user-images.githubusercontent.com/31027858/90639771-cba1ec80-e24c-11ea-94d9-917cb352816b.jpg)

Reply:
On another note, you might find this important.

![Screenshot_20200819-185237261](https://user-images.githubusercontent.com/31027858/90640135-3eab6300-e24d-11ea-9718-6438f39765ed.jpg)

Reply:
Hm, don't know. Looks like `FineLocationManager.removeUpdates()` is not called, but it is (on onPause and onStop)

Perhaps the `@OnLifecycleEvent` annotations don't work as they are supposed to(?)